streams/src/test/java/org/apache/kafka/streams/integration/StreamsUncaughtExceptionHandlerIntegrationTest.java

```diff
@@ -198,35 +195,6 @@ public void shouldShutdownSingleThreadApplication() throws Exception {
         testShutdownApplication(1);
     }
 
-    @Test
-    public void shouldShutDownClientIfGlobalStreamThreadWantsToReplaceThread() throws Exception {
-        builder.addGlobalStore(
-            new KeyValueStoreBuilder<>(
-                Stores.persistentKeyValueStore("globalStore"),
-                Serdes.String(),
-                Serdes.String(),
-                CLUSTER.time
-            ),
-            inputTopic2,
-            Consumed.with(Serdes.String(), Serdes.String()),
-            () -> new ShutdownProcessor(processorValueCollector)
```

Comment:
Seems we need to change `ShutdownProcessor` to let it implement the new `api.Processor` interface to make this work.